### DISCUSSION TITLE
Can't access story iframe if story is not in sidebar

### DISCUSSION THREAD

**Discussion by dpavloff** [storybookjs/storybook#28087] [Help]
### Summary

I've been trying to setup a custom storybook docs page UI. My idea was to transfer the story navigation into the doc page itself (a kin to TOC parameter in `preview.parameters.docs.toc === true`), and hide the stories from the sidebar, leaving only docs pages.

I achieved this part easily by setting `config.docs.docsMode === true` in main.ts.

I also needed to allow user to transfer to any story in a separate page so they would be able to play around with args. Trying to implement that, I've realised that storybook doesn't recognise story URL if it's not in sidebar. Link functions and components from `@storybook/addon-links` didn't do anything.

Most fascinating is the fact that going to the story URL directly with docsMode on will work, **but if you try to hide/ show sidebar, the app will crash.**

Am I going in the wrong direction here, or is this behaviour intended? Why can't we transfer to single story page with addon-links if it's not shown in sidebar?

I've made a quick reproduction here: https://stackblitz.com/edit/sb-testing-docs-mode?file=.storybook%2Fmain.ts

1. Go to doc page `Main Page`
2. Click on link "Click me to go to button--primary!"

Expected result: User transfers to Button -> Primary story page.
Actual result: Nothing happens.

Everything works as expected by settings docsMode: false.

### Create a reproduction

https://stackblitz.com/edit/sb-testing-docs-mode?file=.storybook%2Fmain.ts

**Comment by Marklb** [comment]
I haven't looked at how that routing works in a long time, so I may not be exactly correct, but this seems expected based on how I think of docs mode. I do see what you mean though about the url working, when you navigate to it directly, but it doesn't seem like it is supposed to work. The story renders, but the addon panels area empty. So, they can't play with the controls. That does seem like something that should be fixed to either redirect to where it thinks you most likely want to be in the docs or what I think you want, where the addons would be active.

When saying that you want docs mode, I see that as saying you do not want the individual story UI. It sounds like you want a partial docs mode that only hides the sidebar items. There may be way to do what you want, but I am not sure how. A variation of the workaround from #24390 may temporarily give what you want:
```css
.sidebar-item[data-nodetype="story"] {
  display: none;
}
```

You mention not being able to access the story iframe, but I don't think that wording is what you actually mean. It sounds like you want the full UI that is used for a single story.

When I hear story iframe, I assume you are referring to the following iframe that only contains the story: 
![image](https://github.com/storybookjs/storybook/assets/8348647/e4cb3bd8-36b9-4892-80d2-477e0887ae0d)

Based on your description, the following element is what I now assume you are calling the story iframe, but the bottom panel is part of the manager UI: 
![image](https://github.com/storybookjs/storybook/assets/8348647/54798810-eef6-474e-94d5-f3c24655fa80)

**Reply by dpavloff** [reply]
Thanks for the reply @Marklb! 

You are correct - my wording is a bit off here, by "iframe" I probably meant "story view"(?), a page with controls for a concrete story.

You gave me a good clue that we could hide the story tabs with CSS, the only thing that would make this feature complete is hiding the story results from the search tab (sidebar should not include any story pages), but for now I don't see any clear html difference for story search results and folders/ docs search results.
